pkg/descheduler/descheduler.go

```diff
@@ -59,10 +59,10 @@ func Run(rs *options.DeschedulerServer) error {
 		glog.V(1).Infof("node list is empty")
 		return nil
 	}
-
-	strategies.RemoveDuplicatePods(rs, deschedulerPolicy.Strategies["RemoveDuplicates"], evictionPolicyGroupVersion, nodes)
-	strategies.LowNodeUtilization(rs, deschedulerPolicy.Strategies["LowNodeUtilization"], evictionPolicyGroupVersion, nodes)
-	strategies.RemovePodsViolatingInterPodAntiAffinity(rs, deschedulerPolicy.Strategies["RemovePodsViolatingInterPodAntiAffinity"], evictionPolicyGroupVersion, nodes)
+	nodePodCount := strategies.InitializeNodePodCount(nodes)
+	strategies.RemoveDuplicatePods(rs, deschedulerPolicy.Strategies["RemoveDuplicates"], evictionPolicyGroupVersion, nodes, nodePodCount)
```

Comment:
Given the various strategies operate over the same set of nodes, it would make sense to create a common structure e.g.:
```go
// Victor the Evictor!!!
type Evictor struct {
    ds *options.DeschedulerServer
    nodes []*v1.Node
    nodePodEvictedCount map[*v1.Node]int
    evictionPolicyGroupVersion string
   ...
}
```

that will share all common data. Then you would just have

```go
func (Evictor *evictor) RemoveDuplicatePods(strategy api.DeschedulerStrategy) {...}
func (Evictor *evictor) LowNodeUtilization(strategy api.DeschedulerStrategy) {...}
...
```

cause the `nodePodCount` should bound all the strategies unless we want to support create a classes of node sets, each with its own `nodePodCount`.

Reply:
actually I have been interested in doing something similar since number of strategies increased from one.

Reply:
Yes, I started couple of refactors for this and for registering strategies but couldn't complete them. Let's create a separate issue to track this.